lib/modules/platform/scm-manager/scm-manager-helper.ts

```diff
@@ -0,0 +1,174 @@
+import { ScmManagerHttp } from '../../../util/http/scm-manager';
+import {
+  DefaultBranchSchema,
+  PagedPullRequestSchema,
+  PagedRepoSchema,
+  PullRequestSchema,
+  RepoSchema,
+  UserSchema,
+} from './schema';
+import type { Link, PullRequest, Repo, User } from './schema';
+import type { PullRequestCreateParams, PullRequestUpdateParams } from './types';
+
+let token: string;
+export const setToken = (newToken: string): void => {
+  token = newToken;
+};
+
+export const scmManagerHttp = new ScmManagerHttp();
+
+const URLS = {
+  ME: 'me',
+  ALL_REPOS: 'repositories?pageSize=1000000',
+  REPO: (repoPath: string) => `repositories/${repoPath}`,
+  PULLREQUESTS: (repoPath: string) => `pull-requests/${repoPath}`,
+  PULLREQUESTS_WITH_PAGINATION: (repoPath: string) =>
+    `pull-requests/${repoPath}?status=ALL&pageSize=1000000`,
+  PULLREQUEST_BY_ID: (repoPath: string, id: number) =>
+    `pull-requests/${repoPath}/${id}`,
+};
+
+const CONTENT_TYPES = {
+  ME: 'application/vnd.scmm-me+json;v=2',
+  REPOSITORY: 'application/vnd.scmm-repository+json;v=2',
+  REPOSITORIES: 'application/vnd.scmm-repositoryCollection+json;v=2',
+  GIT_CONFIG: 'application/vnd.scmm-gitDefaultBranch+json;v=2',
+  PULLREQUEST: 'application/vnd.scmm-pullRequest+json;v=2',
+  PULLREQUESTS: 'application/vnd.scmm-pullRequestCollection+json;v=2',
+};
+
+export async function getCurrentUser(): Promise<User> {
+  const response = await scmManagerHttp.getJson(
+    URLS.ME,
+    {
+      scmmContentType: CONTENT_TYPES.ME,
+      token,
+    },
+    UserSchema,
+  );
+  return response.body;
+}
+
+export async function getRepo(repoPath: string): Promise<Repo> {
+  const response = await scmManagerHttp.getJson(
+    URLS.REPO(repoPath),
+    {
+      scmmContentType: CONTENT_TYPES.REPOSITORY,
+      token,
+    },
+    RepoSchema,
+  );
+  return response.body;
+}
+
+export async function getAllRepos(): Promise<Repo[]> {
+  const response = await scmManagerHttp.getJson(
+    URLS.ALL_REPOS,
```

Comment:
is it possible to filter to only get my pr's?

Reply:
We should use it by default then. And only fetch all PRs incase ignorePrAuthor is set to true (this can be added later)